### ISSUE TITLE
Release notes webview shown detached from its parent window

### ISSUE THREAD

**Issue by SebAubin** [vslavik/winsparkle#252]
Hi i'm having trouble with the release notes UI:

![image](https://user-images.githubusercontent.com/11835253/234069653-69c42d93-0216-4b83-86a6-e73b46e56c94.png)

The installation is from a MSIX, could it be releated to that?

When 'im building it in Debug mode, it's correctly aligned:

![image](https://user-images.githubusercontent.com/11835253/234070096-97279def-7d0a-47cb-8a18-955459437fa8.png)

Is anyone is having this issue?

**Comment by WinTDoan** [comment]
Perhaps I can add to this. I was using 0.7.0, but just updated to 0.8.0 to make use of one of the new features. While with 0.7.0 the Release Notes were displayed in the UI dialog, with 0.8.0 they come up in a separate browser window, while the UI dialog text zone remains blank. (The UI dialog initially displays and a new browser window immediately pops up over it).

**Comment by vslavik** [comment]
> Perhaps I can add to this.

You're not really adding much, though. Neither you nor the OP provide any reproduction instructions, provide no information about the Windows version in question (kinda basic), whether you reproduce it in the included sample app or not, whether Edge or MSIE webview is used (right click in it) and so on. Please do, and please also include as much information about the hosting application (framework/runtime/etc) as possible.

**Comment by vslavik** [comment]
> as default browsers

I didn't ask about *that*. I asked you about the *webview in WinSparkle*. I also asked multiple other questions that you didn't answer, including the behavior of the example app, the framework in use, more details about everything etc.

Look, you're presumably a developer. You know how to debug things, or at least how to provide relevant diagnostis details (and if you don't, there's a link above explaining the concepts). So can you please do it? You guys see the issue and are in a position to debug it. I can't reproduce it and simply can't do much until I can.

In particular, @SebAubin has two builds, one with the issue, one without, and can diff them ("Debug", i.e. presence of debug symbols, is not going to be the relevant difference).

**Comment by WinTDoan** [comment]
Sorry to be so dense, but I have no idea what "webview in 
WinSparkle" means. It's not in the docs. If I right-click on the 
(empty) Release Notes field in the UI dialog it gives me nothing 
particularly interesting (screen shot attached). The inspector gives 
the code for an empty window.

At this point, the only thing I can point to is that it worked fine on 
0.7.0 (on two different products) and doesn't work on 0.8.0 on 
multiple computers and on multiple products with multiple default 
browsers and debug vs run vs stand-alone runs in Visual Studio.

The PSDK example *does* seem to work when I compile it using 
the same version of Visual Studio. The only obvious difference in 
the sparkle handling was that it (the example) included a 
win_sparkle_set_http_header, but deleting that had no effect. I can do 
a deeper dive into that, but it would help if I had some idea of what 
might actually be causing a problem. So far as I can tell, I am doing the 
exact same steps.


> 
> > as default browsers
> 
> I didn't ask about *that*. I asked you about the *webview in
> WinSparkle*. I also asked multiple other questions that you didn't
> answer, including the behavior of the example app, the framework in
> use, more details about everything etc.
> 
> Look, you're presumably a developer. You know how to debug things, or
> at least how to provide relevant diagnostis details (and if you don't,
> there's a link above explaining the concepts). So can you please do
> it? You guys see the issue and are in a position to debug it. I can't
> reproduce it and simply can't do much until I can.
> 
> In particular, @SebAubin has two builds, one with the issue, one
> without, and can diff them ("Debug", i.e. presence of debug symbols,
> is not going to be the relevant difference).
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/vslavik/winsparkle/issues/252#issuecomment-15234139
> 25 You are receiving this because you commented.
> 
> Message ID: ***@***.***>


---
2717 Harrison St
Evanston, IL 60201
USA



The following section of this message contains a file attachment
prepared for transmission using the Internet MIME message format.
If you are using Pegasus Mail, or any other MIME-compliant system,
you should be able to save it or view it from within your mailer.
If you cannot, please ask your system administrator for assistance.

   ---- File information -----------
     File:  Sparkle Screenshot.png
     Date:  26 Apr 2023, 9:34
     Size:  86934 bytes.
     Type:  Unknown

**Comment by WinTDoan** [comment]
So, after a bit of experimenting, I found that in the appcast XML file, formatting the release notes link as

`		<sparkle:releaseNotesLink>https://estima.com/patches/rats/RATS1000Updates.shtml</sparkle:releaseNotesLink>
`
works (is presented in the release notes box) but as

```
		<sparkle:releaseNotesLink>
                  https://estima.com/patches/rats/RATS1000Updates.shtml
                 </sparkle:releaseNotesLink>
```

(for legibility) doesn't (presents it in a separate browser window) in 0.8.0. (0.7.0 didn't seem to care)

**Comment by vslavik** [comment]
*(Happily deleting 30 minutes worth of already written detailed reply...)*

> So, after a bit of experimenting, I found that in the appcast XML file, formatting the release notes link as
...
> (for legibility) doesn't (presents it in a separate browser window) in 0.8.0. (0.7.0 didn't seem to care)

This implies that the answer to which control you use is WebView2/Edge, which 0.7.0 didn't support.

It is not shown in "a separate browser window" — this is an embedded WebView control, not a browser window and it is not separate one. It's the same one, but the ActiveX control failed to be reparented into its correct position and so is shown at position (0,0) under the root HWND (the desktop).

That WebView2 would react *that* way to a bad input is truly out of the left field. I'd never guess that 😳

Can you please confirm that in the included example app? [This](https://github.com/vslavik/winsparkle/blob/10aa322213d1b074bf47f5ab484ae7009ac6303b/examples/psdk/app_psdk.rc#L13) is where the feed URL is set. I'll do the same here tomorrow too to confirm.

Whitespace is significant in XML and spaces are not valid characters in URL, so your "for legibility" `<sparkle:releaseNotesLink>` value *is* invalid URL and the feed is malformed. But *if* this is the cause and such malformed links are common (3 people affected here already...), then there's little harm in fixing it in WinSparkle I suppose.

**Comment by dion-gionet** [comment]
Hey @vslavik ! I'm working with @SebAubin on integrating WinSparkle on our app. The original issue appears to have been because of the old webview as we were using winsparkle 0.7.0. 

After upgrading to 0.8.0, we encountered the same issue as @WinTDoan as our releaseNotesLink also contained whitespaces for legibility as we use both Sparkle and WinSparkle and it worked fine on Sparkle. 

We removed the whitespaces as suggested and the webview now works as expected.
